scripts/src/admin/services.ts

```diff
@@ -546,12 +554,12 @@ export async function changeOffer(body: Partial<Offer>, params: { offer_id: stri
 
 	let didChange = false;
 	if (body && body.cap) {
-		if (body.cap.total && !isNothing(parseInt(body.cap.total as any, 10)) && parseInt(body.cap.total as any, 10) >= 0) {
-			offer.cap.total = parseInt(body.cap.total as any, 10);
+		if (!isNothing(body.cap.total)  && body.cap.total >= 0) {
+			offer.cap.total = body.cap.total;
 			didChange = true;
 		}
-		if (body.cap.per_user && !isNothing(parseInt(body.cap.per_user as any, 10)) && parseInt(body.cap.per_user as any, 10) >= 0) {
-			offer.cap.per_user = parseInt(body.cap.per_user as any, 10);
+		if (!isNothing(body.cap.per_user) && body.cap.per_user >= 0) {
```

Comment:
What's the meaning of `body.cap.per_user = 0` is it no cap or a cap 0 so no offers at all?
If it's no cap then you can do `if (body.cap.per_user) {`

Reply:
it means no offer will return to any user. 0 is a valid option for cap.